crates/pathfinder/examples/re_execute.rs

```diff
@@ -0,0 +1,194 @@
+use std::num::NonZeroU32;
+
+use anyhow::Context;
+use mimalloc::MiMalloc;
+use pathfinder_common::{BlockNumber, BlockTimestamp, ChainId, SequencerAddress};
+use pathfinder_rpc::cairo::starknet_rs::ExecutionState;
+use pathfinder_storage::{BlockId, JournalMode, Storage};
+use primitive_types::U256;
+use starknet_gateway_types::reply::transaction::{Receipt, Transaction};
+
+#[global_allocator]
+static GLOBAL: MiMalloc = MiMalloc;
```

Comment:
Because it significantly improves re_execute performance.

With the standard (glibc) allocator:

```
Benchmark 1: ./target/release/examples/re_execute ./testnet2.sqlite 123000 124000
  Time (mean ± σ):      5.532 s ±  0.110 s    [User: 127.095 s, System: 0.829 s]
  Range (min … max):    5.422 s …  5.767 s    10 runs
```

With mimalloc:

```
Benchmark 1: ./target/release/examples/re_execute ./testnet2.sqlite 123000 124000
  Time (mean ± σ):      4.068 s ±  0.049 s    [User: 92.965 s, System: 0.921 s]
  Range (min … max):    4.000 s …  4.127 s    10 runs
```

Reply:
The difference used to be much larger, but that was mainly due to starknet_in_rust doing dumb things causing excessive cloning of JSON Value trees.

Reply:
Maybe just add a comment for that 👍

Reply:
Done.